### DISCUSSION TITLE
Allow pasting joined lines not to create empty lines

### DISCUSSION THREAD

**Discussion by eugenesvk** [helix-editor/helix#10138] [General]
- create two cursors with with `copy_selection_on_next_line`
- extend to selection to capture the whole line with the convenient `extend_to_line_bounds` (in the screenshot the `extend_to_line_end` command is just as convenient and doesn't capture newlines, but too often it's not as you have cursors from e.g. a search operation in the middle of a line, so extending is better than moving and selecting, and also I don't get the logic why the extra newlines would appear)
- copy both lines with `yank_joined_to_clipboard`
- paste with `paste_clipboard_after`
- see two extra empty lines created, though you'd expect the joined two lines to only have two newlines between them

Quick illustration 
![helix](https://user-images.githubusercontent.com/12956286/151656447-dd68a599-263d-4f8b-9ea1-2702e4dfaa2f.gif)

**Comment by dead10ck** [comment]
I believe this is related to / duplicating https://github.com/helix-editor/helix/issues/1377. The issue is that the selection includes the newline characters at the end of the line, which you are inserting *after* the last newline under your cursor. I'd guess that shift-p would not do this.

**Comment by eugenesvk** [comment]
The other issue is about the first newline before the paste, which is not what I write about — my issue is with the two newlines that are created AFTER each line, and `paste_clipboard_before` doesn't change that. For some reason joining multiple selections doesn't just glue them, but inserts newlines in between (or maybe it happens on paste, not on join)
